### DISCUSSION TITLE
datasets/pose/dog-pose/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17946] [Docs]
# datasets/pose/dog-pose/

Discover the Dog-Pose dataset for pose detection. Featuring 6,773 training and 1,703 test images, it's a robust dataset for training YOLO11 models.

https://docs.ultralytics.com/datasets/pose/dog-pose/

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing this resource with the community! 🚀 The Dog-Pose dataset looks fantastic and is sure to be a valuable asset for those exploring pose detection models like YOLO.

We recommend users interested in the dataset check it out directly in the [Dog-Pose dataset documentation](https://docs.ultralytics.com/datasets/pose/dog-pose/). For additional details on how to train YOLO models with custom datasets, refer to our [Model Training Guide](https://docs.ultralytics.com/guides/model-training-tips/).

If you're exploring pose detection tasks or facing any challenges, feel free to ask questions and include as much context or data as possible. If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can assist effectively.

For real-time support and discussions, consider joining the Ultralytics community:
- Chat with us on [Discord](https://discord.com/invite/ultralytics) 🎧
- Discuss on [Discourse](https://community.ultralytics.com) 📚
- Explore shared knowledge on our [Subreddit](https://reddit.com/r/Ultralytics) 💡

Stay updated by upgrading to the latest `ultralytics` package:

```bash
pip install -U ultralytics
```

You can also try YOLO in various verified environments for ease of use:
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

An Ultralytics engineer will review and engage with this post soon. Thank you for contributing to the Ultralytics community! 🙌

**Comment by chengziNaomi** [comment]
When we annotate the key point data set in a picture, if the two boxes have overlapping parts, how to annotate the key points in the overlapping boxes?

**Reply by glenn-jocher** [reply]
@chengziNaomi when annotating overlapping bounding boxes in a keypoint dataset, ensure that each object's keypoints are labeled independently, even if they share overlapping regions. Annotate each keypoint based on its respective object, maintaining clear separation to avoid ambiguity during model training. For guidelines on keypoint annotation, refer to [Dog-Pose Dataset Documentation](https://docs.ultralytics.com/datasets/pose/dog-pose/).

**Reply by chengziNaomi** [reply]
If I have multiple objects in a picture, should I frame all the objects at once and then label the key points together, or should I frame an object, label the key points of that object, then frame an object, and then label the key points of that object?

**Reply by glenn-jocher** [reply]
@chengziNaomi when annotating a single image with multiple objects, you should annotate each object separately. Frame one object, label its keypoints completely, then move to the next object. This ensures clarity and avoids confusion during model training. Refer to [Dog-Pose Dataset Documentation](https://docs.ultralytics.com/datasets/pose/dog-pose/) for more annotation insights.

**Reply by chengziNaomi** [reply]
Thank you very much! and how can I draw heatmap for key point detection?

**Reply by glenn-jocher** [reply]
You're welcome! To draw a heatmap for keypoint detection, you can preprocess the keypoints into Gaussian blobs on a blank image using libraries like OpenCV or NumPy, then overlay them on the original image. For detailed implementation, check external heatmap solutions or visualization tools as it currently falls outside YOLO's built-in features. Let me know if you have further questions!

**Comment by Peakednesss** [comment]
I can just find 20 points, 4 for ears, 1 for nose, 1 for mouth, 2 for tail, and 12 for legs, where are the other four points?

**Reply by glenn-jocher** [reply]
@Peakednesss the Dog-Pose dataset includes 24 keypoints. Beyond the 20 you identified (ears, nose, mouth, tail, and legs), the remaining four points likely represent additional key body landmarks like the torso or other specific regions defined for pose estimation. For detailed annotations, refer to the dataset YAML file: https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/dog-pose.yaml.

**Reply by Peakednesss** [reply]
thank you very much

**Reply by glenn-jocher** [reply]
You're welcome! The credit goes to the YOLO community and the Ultralytics team for their efforts in making these resources accessible. Let us know if you have further questions!

**Reply by goodbetterbest1** [reply]
do you know where are the other 4 points’ definitions? I can’t find any useful information in the yaml document. thx
